django_elasticsearch_dsl/management/commands/search_index.py

```diff
@@ -139,13 +153,85 @@ def _delete(self, models, options):
             index.delete(ignore=404)
         return True
 
+    def _update_alias(self, es_conn, alias, new_index, alias_exists, options):
+        alias_actions = [{"add": {"alias": alias, "index": new_index}}]
+        old_indices = []
+        alias_delete_actions = []
+        if alias_exists:
+            # Elasticsearch will return an error if we search for
+            # indices by alias but the alias doesn't exist. Therefore,
+            # we want to be sure the alias exists.
+            old_alias_indices = es_conn.indices.get_alias(name=alias)
+            old_indices = list(old_alias_indices.keys())
+            alias_actions.append(
+                {"remove": {"alias": alias, "indices": old_indices}}
+            )
+            alias_delete_actions = [
+                {"remove_index": {"index": index}} for index in old_indices
+            ]
+
+        es_conn.indices.update_aliases({"actions": alias_actions})
+        self.stdout.write(
+            "Added alias '{}' to index '{}'".format(alias, new_index)
+        )
+
+        if old_indices:
+            if len(old_indices) == 1:
+                stdout_term = "index"
+            else:
+                stdout_term = "indices"
+
+            old_indices_str = ", ".join(old_indices)
+            self.stdout.write(
+                "Removed alias '{}' from {} '{}'".format(
+                    alias, stdout_term, old_indices_str
+                )
+            )
+            if alias_delete_actions and not options['atomic_no_delete']:
+                es_conn.indices.update_aliases(
+                    {"actions": alias_delete_actions}
+                )
+                self.stdout.write(
+                    "Deleted {} '{}'".format(stdout_term, old_indices_str)
+                )
+
     def _rebuild(self, models, options):
-        if not self._delete(models, options):
+        if not options['atomic'] and not self._delete(models, options):
             return
 
+        if options['atomic']:
+            alias_index_pairs = []
+            index_suffix = "-" + datetime.now().strftime("%Y%m%d%H%M%S%f")
+            for index in registry.get_indices(models):
+                # The alias takes the original index name value. The
+                # index name sent to Elasticsearch will be the alias
+                # plus the suffix from above.
+                new_index = index._name + index_suffix
```

Comment:
There are limit for [index name](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html#indices-create-api-path-params) 255 charector. Can you limit slice the string so it never exceeds 255 charector?

Reply:
Yes, just committed the changes!